src/Raven.Studio/typescript/models/database/timeSeries/timeSeriesEntryModel.ts

```diff
@@ -0,0 +1,204 @@
+                return `The maximum number of possible rollup values (${timeSeriesEntryModel.numberOfPossibleRollupValues}) has been reached.`;
+            }
+            
+            if (!this.isRollupEntry() && this.values().length === timeSeriesEntryModel.numberOfPossibleValues) {
+                return `The maximum number of possible values (${timeSeriesEntryModel.numberOfPossibleValues}) has been reached.`;
+            }
+
+            return "";
+        });
+    }
+
+    addValue() {
+        if (this.isRollupEntry()) {
+            const newRollupData = new rollupDataModel(); // todo check..
+            this.rollupValues.push(newRollupData);
+        } else {
+            const newValue = new timeSeriesValue(0);
+            this.values.push(newValue);
+        }
+    }
+    
+    removeValue(value: timeSeriesValue) {
+        this.values.remove(value);
+    }
+
+    removeRollupData(rollup: rollupDataModel) {
+        this.rollupValues.remove(rollup);
+    }
+    
+    private initValidation() {
+        this.name.extend({
+            required: true,
+            validation: [
+                {
+                    validator: () =>  this.isRollupEntry() || !this.canEditName || !this.name().includes("@"),
+                    message: "A Time Series name cannot contain '@'. This character is reserved for Time Series Rollups."
+                }
+            ]
+        });
+        
+        this.timestamp.extend({
+            required: true,
+            validation: [
+                {
+                    validator: () => this.timestamp().isValid(),
+                    message: "Please enter a valid date"
+                }
+            ]
+        });
+
+        this.tag.extend({
+            maxLength: 255
+        });
+        
+        this.values.extend({
+            validation: [
+                {
+                    validator: () => this.isRollupEntry() || this.values().length > 0,
+                    message: "At least one value is required"
+                }
+            ]
+        });
+        
+        this.rollupValues.extend({
+            validation: [
+                {
+                    validator: () => !this.isRollupEntry() || this.rollupValues().length > 0,
+                    message: "At least one rollup value is required"
+                }
+            ]
+        });
+        
+        this.validationGroup = ko.validatedObservable({
+            name: this.name,
+            tag: this.tag,
+            timestamp: this.timestamp,
+            values: this.values,
+            rollupValues: this.rollupValues
+        });
+    }
+    
+    private flattenRollupValues() {
+        return this.rollupValues().reduce((result: number[], next: rollupDataModel) => {
+            addToResult(result, next);
+            return result;
+        }, []);
+
+        function addToResult(result: number[], item: rollupDataModel) {
```

Comment:
move to rollupDataModel, create function getValues(): number[] { }

Reply:
result.push(...item.getValues())

Reply:
done

Reply:
do we need this one liner function? 

I think we can simply do: result.push(...item.getValues());

Reply:
done + fixed spaces